lib/ansible/modules/cloud/xenserver/xenserver_guest.py

```diff
@@ -0,0 +1,1241 @@
+
+            disk_units = dict(tb=4, gb=3, mb=2, kb=1, b=0)
+
+            if unit in disk_units:
+                return int(size * (1024 ** disk_units[unit]))
+            else:
+                self.module.fail_json(msg="VM check disks[%s]: '%s' is not a supported unit for disk size. Supported units are ['%s']." %
+                                      (position, unit, "', '".join(sorted(disk_units.keys(), key=lambda key: disk_units[key]))))
+        else:
+            return None
+
+    @staticmethod
+    def get_cdrom_type(vm_cdrom_params):
+        # TODO: implement support for detecting type host. No server to test
+        # this on at the moment.
+        if vm_cdrom_params['empty']:
+            return "none"
+        else:
+            return "iso"
+
+    @staticmethod
+    def is_valid_mac_addr(mac_addr):
+        """
+        Function to validate MAC address for given string
+        Args:
+            mac_addr: string to validate as MAC address
+
+        Returns: (Boolean) True if string is valid MAC address, otherwise False
+        """
+        mac_addr_regex = re.compile('[0-9a-f]{2}([-:])[0-9a-f]{2}(\\1[0-9a-f]{2}){4}$')
+        return bool(mac_addr_regex.match(mac_addr))
+
+
+def main():
+    argument_spec = xenserver_common_argument_spec()
+    argument_spec.update(
+        state=dict(type='str', default='present',
+                   choices=['present', 'absent', 'poweredon']),
+        name=dict(type='str', aliases=['name_label']),
+        name_desc=dict(type='str'),
+        uuid=dict(type='str'),
+        template=dict(type='str', aliases=['template_src']),
+        template_uuid=dict(type='str'),
+        is_template=dict(type='bool', default=False),
+        folder=dict(type='str'),
+        hardware=dict(
+            type='dict',
+            options=dict(
+                num_cpus=dict(type='int'),
+                num_cpu_cores_per_socket=dict(type='int'),
+                memory_mb=dict(type='int'),
+            ),
+            default=dict(),
+        ),
+        disks=dict(
+            type='list',
+            elements='dict',
+            options=dict(
+                size=dict(type='str'),
+                size_tb=dict(type='str'),
+                size_gb=dict(type='str'),
+                size_mb=dict(type='str'),
+                size_kb=dict(type='str'),
+                size_b=dict(type='str'),
+                name=dict(type='str', aliases=['name_label']),
+                name_desc=dict(type='str'),
+                sr=dict(type='str'),
+                sr_uuid=dict(type='str'),
+            ),
+            default=[],
+            aliases=['disk'],
+            mutually_exclusive=[
+                ['size', 'size_tb', 'size_gb', 'size_mb', 'size_kb', 'size_b'],
+            ],
+        ),
+        cdrom=dict(
+            type='dict',
+            options=dict(
+                type=dict(type='str', required=True, choices=['none', 'host', 'iso']),
+                iso=dict(type='str'),
```

Comment:
That cryptic is error is coming from here. You can rename this parameter as `iso_image` or something similar which makes easy to understand for end user and will resolve the error too.